helpers/helper-translation-discussion.php

```diff
@@ -547,9 +564,34 @@ function( $comment_fields ) {
 			}
 		);
 
+		add_filter( 'jetpack_mentions_should_load_ui', '__return_true' );
+
 		remove_action( 'comment_form_top', 'rosetta_comment_form_support_hint' );
 
-		$post = self::maybe_get_temporary_post( self::get_shadow_post_id( $this->data['original_id'] ) );
+		$post                       = self::maybe_get_temporary_post( self::get_shadow_post_id( $this->data['original_id'] ) );
+		$validator_email_addresses  = self::get_validator_email_addresses_for_original_id( $this->data['locale_slug'], $this->data['original_id'] );
+		$commenters_email_addresses = array_values( GP_Notifications::get_commenters_email_addresses( $comments, $validator_email_addresses ) );
+
+		$all_email_addresses = array_merge(
+			$validator_email_addresses,
+			$commenters_email_addresses
+		);
+
+		$users         = array_map(
+			function( $email ) {
+				$user = get_user_by( 'email', $email );
+				return array(
+					'ID'            => $user->ID,
+					'user_login'    => $user->user_login,
+					'user_nicename' => $user->user_nicename,
+					'display_name'  => '',
+					'source'        => array( 'translators' ),
+					'image_URL'     => get_avatar_url( $user->ID ),
+				);
+			},
+			$all_email_addresses
+		);
+		$mentions_list = $users;
```

Comment:
To avoid the .org dependency, we could do something like this (please think of a better filter name)

```suggestion
		$mentions_list = apply_filters( 'glotpress_comment_mentions', array() );
```

And then in `WPorg_GlotPress_Notifications` we can use `add_filter()` to supply those e-mail addresses from there.